docs/source/users/index.md

```diff
@@ -958,6 +958,19 @@ configuration.
 jupyter lab --AiExtension.allowed_providers=openai --AiExtension.allowed_providers=ai21
 ```
 
+### Chat memory size
+
+This configuration allows for setting the number of chat exchanges the model
+uses as context when generating a response.
+
+One chat exchange corresponds a user message and its AI response.
+E.g. Setting value of 4 means the model will use the last 8 messages total.
+Default value is 2.
```

Comment:
Reword as follows: 

One chat exchange corresponds to a user query message and its AI response, which counts as two messages. `k` denotes one chat exchange, i.e., two messages.  The default value of `k` is 2, which corresponds to 4 messages.

Reply:
@michaelchia This is a substantive PR, thanks for the huge effort. 

I think it may be better to rename `llm_chat_history` to `llm_chat_memory` to avoid confusion with the chat history itself. 

I was trying to make sure that `llm_chat_history` retains only the default (`k=2`) messages, so to test this, I added the following logging lines of code to `clear.py` just before we clear the memory. 
<img width="982" alt="image" src="https://github.com/user-attachments/assets/779589e6-f8b8-41f7-994c-583cbb0eb825">

Then, I asked the LLM five questions 
<img width="1574" alt="image" src="https://github.com/user-attachments/assets/3e8939df-9aa3-4796-b4ce-98066fd9d8cd">

and then called `/clear`. The `self.llm_chat_history` shows all five chat exchanges not just two. 
<img width="1238" alt="image" src="https://github.com/user-attachments/assets/45fa4fe9-2c3b-4a59-a055-89a723c79f11">

Calling `/clear` again shows the memory has been cleared, so that part is verified. 
<img width="1240" alt="image" src="https://github.com/user-attachments/assets/87cf6cc2-a715-4154-9832-6356d6e0e837">

I am able to verify that `/clear` also removes chat history and that full chat history is retained even though conversational memory is only intended to be retained as `k` exchanges.

Reply:
`all_messages` is the unfiltered list. `messages` and `aget_messages` (which langchain uses) will apply the filter. This is so that if `k` is dymically reconfigured to be larger (e.g. via UI in the future), it has not lost the information about the other messages. Strictly speaking, `all_messages` should be a private variable. I was just having a bit of trouble making it private in pydantic. I'll figure it out and make that change.

Noted on the docs changes. I'll make the changes when I have some time.

Reply:
@michaelchia - That is correct, I verified this after your clarification.